code/globalincs/linklist.h

```diff
@@ -83,4 +83,87 @@ do {												\
 #define NOT_EMPTY(head)		((head)->next != (head))
 #define EMPTY(head)			((head) == nullptr || (head)->next == (head))
 
+template <class T>
+class volition_linked_list
+{
+	T* head;
+
+public:
+	class iterator
+	{
+		T* ptr;
```

Comment:
This can be changed to `const T*` since this class (and the iterator) does not modify the pointer value.

Reply:
updated with a `const_iterator`